addons/website/static/src/snippets/s_popup/000.js

```diff
@@ -25,54 +29,214 @@ const SharedPopupWidget = publicWidget.Widget.extend({
         // `contenteditable=true` attribute in edit mode. It will result in a
         // ugly white bar.
         // tl;dr: this is keeping those 2 elements visibility synchronized.
-        'show.bs.modal': '_onModalShow',
-        'hidden.bs.modal': '_onModalHidden',
-    },
+        "_root": {
+            "t-on-show.bs.modal": () => this.popupShown = true,
+            "t-on-hidden.bs.modal": this.onModalHidden,
+            "t-att-class": () => ({ "d-none": !this.popupShown }),
+        },
+    }
 
-    /**
-     * @override
-     */
-    destroy() {
-        this._super(...arguments);
+    setup() {
+        this.popupShown = false;
+    }
 
+    destroy() {
+        // TODO: support edit mode
         // Popup are always closed when entering edit mode (see PopupWidget),
         // this allows to make sure the class is sync on the .s_popup parent
         // after that moment too.
-        if (!this.editableMode) {
-            this.el.classList.add('d-none');
+        // if (!this.editableMode) {
+            this.popupShown = false;
```

Comment:
Note that upon destroy, the registered cleanups have already been called, and updating the variable will not have any visual impact. Maybe the whole destroy method can be removed ?